generators/typescript/sdk/generator/src/TemplateGenerator.ts

```diff
@@ -258,14 +260,16 @@ export class TemplateGenerator {
         location,
         wireOrOriginalName,
         nameBreadcrumbs,
-        indentationLevel
+        indentationLevel,
+        isUnionType
     }: {
         otd: ObjectTypeDeclaration;
         name: string | undefined;
         location: FdrSnippetTemplate.PayloadLocation;
         wireOrOriginalName: string | undefined;
         nameBreadcrumbs: string[] | undefined;
         indentationLevel: number;
+        isUnionType: boolean;
```

Comment:
Another nit: you could call this `isObjectInlined` instead of `isUnionType` and that could add some clarity as to what's happening (in addition to a comment)